firmware/drivers/citiroc/citiroc.c

```diff
@@ -87,6 +87,23 @@ void citiroc_daq_set_hvps_curr(uint16_t curr)
 	CITIROC->HVPSR |= ((curr & 0xffff)<<16);
 }
 
+void citiroc_calib_set(uint32_t calibration)
+{
+	uint32_t temp;
+	temp = calibration & CALIBPER;
+
+	/*For period greater than zero, apply the register, else, disable pulse*/
+	if (temp != 0)
+	{
+		CITIROC->ASICCALR &= ~(0xffffffff);
```

Comment:
You could move this assignment above the `if` statement and avoid the `else` clause altogether.